src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -612,42 +650,61 @@ void Hole::updateHoleCutParams()
         return;
     }
 
+    // get diameter and size
+    double diameterVal = Diameter.getValue();
+
+    // handle thread types
     std::string threadType = ThreadType.getValueAsString();
     if (threadType == "ISOMetricProfile" || threadType == "ISOMetricFineProfile") {
         if (ThreadSize.getValue() < 0) {
             throw Base::IndexError("Thread size out of range");
             return;
         }
 
-        // get diameter and size
-        double diameter = threadDescription[ThreadType.getValue()][ThreadSize.getValue()].diameter;
         std::string threadSize{ ThreadSize.getValueAsString() };
 
         // we don't update for these settings but we need to set a value for new holes
+        // furthermore we must assure the hole cut diameter is not <= the hole diameter
         // if we have a cut but the values are zero, we assume it is a new hole
         // we take in this case the values from the norm ISO 4762 or ISO 10642
         if (holeCutType == "Counterbore") {
             // read ISO 4762 values
             const CutDimensionSet& counter = find_cutDimensionSet(threadType, "ISO 4762");
             const CounterBoreDimension& dimen = counter.get_bore(threadSize);
-            if (HoleCutDiameter.getValue() == 0.0) {
-                HoleCutDiameter.setValue(dimen.diameter);
-                HoleCutDepth.setValue(dimen.depth);
+            if (HoleCutDiameter.getValue() == 0.0 || HoleCutDiameter.getValue() <= diameterVal) {
+                // there is no norm defining counterbores for all sizes, thus we need to use the
+                // same fallback as for the case HoleCutTypeMap.count(key)
+                if (dimen.diameter != 0.0) {
+                    HoleCutDiameter.setValue(dimen.diameter);
+                    HoleCutDepth.setValue(dimen.depth);
+                }
+                else {
+                    // valid values for visual feedback
+                    HoleCutDiameter.setValue(Diameter.getValue() + 0.1);
+                    HoleCutDepth.setValue(0.1);
+                }
             }
             if (HoleCutDepth.getValue() == 0.0)
                 HoleCutDepth.setValue(dimen.depth);
+            HoleCutCountersinkAngle.setReadOnly(true);
         }
         else if (holeCutType == "Countersink") {
             // read ISO 10642 values
             const CutDimensionSet& counter = find_cutDimensionSet(threadType, "ISO 10642");
-            if (HoleCutDiameter.getValue() == 0.0) {
+            if (HoleCutDiameter.getValue() == 0.0 || HoleCutDiameter.getValue() <= diameterVal) {
                 const CounterSinkDimension& dimen = counter.get_sink(threadSize);
-                HoleCutDiameter.setValue(dimen.diameter);
+                if (dimen.diameter != 0.0) {
+                    HoleCutDiameter.setValue(dimen.diameter);
+                } 
+                else {
+                    HoleCutDiameter.setValue(Diameter.getValue() + 0.1);

```

Comment:
> This will only look good on small bolts. Consider a more general fallback.

Sure and in the comment some lines above I wrote why I see no other way. When there is no norm defining e.g. a countersink screw for M72, we cannot do much. However, at least now we check that the diameter is large enough to avoid calculation errors.

If you know a better fallback, let's please sort this out to a later PR. To keep the overview let's fix the issues in small portions. PR #4343 fixes the diameter check and thus solves an annoying bug. It can later be beautified with a better fallback in case we find one.